### PR TITLE
feat(Microsoft Login): Register and login teacher

### PR DESCRIPTION
**Pull request by hirokiterashima**
## Notes
- This PR introduces a lot of code duplication with Google authentication, but you can ignore them for now. I'll address them in a separate PR to reduce the review scope of this PR.
- Don't worry about styles (e.g. MS button looks like Google button) for now. We'll also address that in a separate PR.

## Changes
- Add Register and Login with Microsoft features for teachers

## Test Prep
- Test with https://github.com/WISE-Community/WISE-API/pull/263

## Test
- Register/Sign in with Microsoft buttons (for teachers only)
   - appear when properties are set in wise-dockerdev.properties.
   - do NOT appear when properties are NOT set in wise-dockerdev.properties.
- Register teacher with Microsoft
   - with new MS account => lets you create new teacher account 
   - with existing MS account => shows page with message "Microsoft user already has a WISE account"
- Login as teacher with Microsoft
   - with already-registered MS account => works
   - with unregistered MS account => shows page with message "Sorry, your Microsoft User ID was not found in WISE."

### CODE REVIEW

File: src/app/register/register-microsoft-user-already-exists/register-microsoft-user-already-exists.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+
```
Comment (geoffreykwan) [inline src/app/register/register-microsoft-user-already-exists/register-microsoft-user-already-exists.component.spec.ts]: This empty line can be deleted.

**Review by geoffreykwan — CHANGES_REQUESTED** [review]
It mostly works. I found one scenario that doesn't work properly though.

If I already have a WISE account that signed up with Microsoft and then I try to register a new teacher account with that same Microsoft account, it brings me to the create teacher account form.

File: src/app/register/register-microsoft-user-already-exists/register-microsoft-user-already-exists.component.spec.ts (hunk 1 above)
Comment (hirokiterashima) [reply src/app/register/register-microsoft-user-already-exists/register-microsoft-user-already-exists.component.spec.ts]: done.

**Comment by hirokiterashima** [comment]
> If I already have a WISE account that signed up with Microsoft and then I try to register a new teacher account with that same Microsoft account, it brings me to the create teacher account form.

Good catch. If you pull the latest from API, this should be resolved. It should show the message "Microsoft user already has a WISE account"

**Review by geoffreykwan — APPROVED** [review]
Looks good.

I noticed the "Microsoft/Google user already has a WISE account" page should have the "Sign in with Microsoft/Google" button center aligned but that can be fixed later.

**Pull request merged**